### PR TITLE
aggiunto esercizio count change

### PR DESCRIPTION
**Pull request by christianmorettin**

### CODE REVIEW

File: src/main/scala/recfun/RecFun.scala [hunk 1]
```diff
@@ -14,15 +14,52 @@ object RecFun extends RecFunInterface {
   /**
    * Exercise 1
    */
-  def pascal(c: Int, r: Int): Int = ???
+  def pascal(c: Int, r: Int): Int = {
+    if (c < 0 || r < 0 )
+        throw new RuntimeException("Le colonne e le righe passate devono essere maggiori di zero")
+    else{
+      if (c == 0 || r==0 || c == r) 1 else pascal(c - 1, r - 1) + pascal(c, r - 1)
```
Comment (tcappellari) [inline src/main/scala/recfun/RecFun.scala]: You can remove the `r==0` condition.

File: src/main/scala/recfun/RecFun.scala [hunk 2]
```diff
@@ -14,15 +14,52 @@ object RecFun extends RecFunInterface {
...
+      }else
+      if (chars.head == '(') {
+        parBalance(chars.tail, parOpen + 1)
+      } else
+      if (chars.head == ')') {
+        parBalance(chars.tail, parOpen - 1)
+      }
+      else{
+        parBalance(chars.tail, parOpen)
+      }
```
Comment (tcappellari) [inline src/main/scala/recfun/RecFun.scala]: Please rewrite using `match`

**Review by tcappellari — CHANGES_REQUESTED** [review]

File: src/main/scala/recfun/RecFun.scala (hunk 1 above)
Comment (christianmorettin) [reply src/main/scala/recfun/RecFun.scala]: Fatto

File: src/main/scala/recfun/RecFun.scala [hunk 3]
```diff
@@ -31,19 +31,16 @@ object RecFun extends RecFunInterface {
...
-      else{
-        parBalance(chars.tail, parOpen)
+      else if (chars.isEmpty) {
+        parOpen == 0
       }
+      else
+      chars.head match  {
+        case '(' => parBalance(chars.tail, parOpen + 1)
+        case ')' => parBalance(chars.tail, parOpen - 1)
+        case _ => parBalance(chars.tail, parOpen)
```
Comment (tcappellari) [inline src/main/scala/recfun/RecFun.scala]: Okay, or:
```
chars match {
      case Nil => ...
      case '(' :: xs => ...
      case ')' :: xs => ...
      case _ :: xs => ...
```

**Review by tcappellari — APPROVED** [review]

**Pull request merged**